### PR TITLE
update readme

### PR DESCRIPTION
**Pull request by gabrielittner**
- tries to go through the steps in the right order
- mention all properties we support
- mention Sonatype OSS more explicitly
- make signing more prominent

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,63 +21,110 @@ apply plugin: "com.vanniktech.maven.publish"
...
-Can be found [here](https://oss.sonatype.org/#nexus-search;quick~gradle-maven-publish-plugin). Current one is:
+### Setting properties
 
-```groovy
-classpath 'com.vanniktech:gradle-maven-publish-plugin:0.9.0-SNAPSHOT'
+To configure the coordinates of your published artifacts as well as the POM this plugin 
+uses Gradle properties. It's generally recommended to set them in your `gradle.properties`
+file.
+
+There are 3 required properties:
```
Comment (vanniktech) [inline README.md]:
```suggestion
There are three required properties:
```

File: README.md [hunk 2]
```diff
@@ -21,63 +21,110 @@ apply plugin: "com.vanniktech.maven.publish"
...
+```
+GROUP=com.test.mylibrary
+POM_ARTIFACT_ID=mylibrary-runtime
+VERSION_NAME=3.0.5
+```
+
+In addition to that there are some optional properties to give more details about your project:
```
Comment (vanniktech) [inline README.md]:
```suggestion
In addition, there are some optional properties to give more details about your project:
```

File: README.md [hunk 3]
```diff
@@ -21,63 +21,110 @@ apply plugin: "com.vanniktech.maven.publish"
...
+the rest by putting them into `<project-dir>/gradle.properties`.
+
+### Where to upload to
+
+Without any further configuration the plugin has two tasks. `uploadArchives` which will upload
+to Sonatype OSS (Maven Central) by default. To publish to the local maven repository on your 
+machine (`~/m2/repository`) there is `installArchives`. 
 
-***Uploading:***
+The username and password for Sonatype OSS can be provided as Gradle properties or environment
```
Comment (vanniktech) [inline README.md]: Let's mention which properties need to be set here for Gradle

File: README.md [hunk 4]
```diff
@@ -21,63 +21,110 @@ apply plugin: "com.vanniktech.maven.publish"
...
+More information can be found in the [Documentation of the Extension](src/main/kotlin/com/vanniktech/maven/publish/MavenPublishPluginExtension.kt)
 
 __Note:__ To prevent looping behavior, especially in Kotlin projects / modules, you need to run the `uploadArchives` task with `--no-daemon`and `--no-parallel` flags:
 
 `./gradlew uploadArchives --no-daemon --no-parallel`
 
-***Releasing:***
+### Signing
+
+The plugin supports setting up all your artifacts to be signed with GPG. Since that is a requirement
```
Comment (vanniktech) [inline README.md]:
```suggestion
The plugin supports signing all of your artifacts with GPG. This is a requirement when publishing to Maven Central - our default behavior. Signing parameters can be configured via:
```

File: README.md [hunk 5]
```diff
@@ -21,63 +21,110 @@ apply plugin: "com.vanniktech.maven.publish"
...
+for our default publishing target Maven Central, signing is enabled by default. For that please add the following
```
Comment (vanniktech) [inline README.md]:
```suggestion

```

File: README.md [hunk 6]
```diff
@@ -21,63 +21,110 @@ apply plugin: "com.vanniktech.maven.publish"
...
 __Note:__ To prevent looping behavior, especially in Kotlin projects / modules, you need to run the `uploadArchives` task with `--no-daemon`and `--no-parallel` flags:
 
 `./gradlew uploadArchives --no-daemon --no-parallel`
 
-***Releasing:***
+### Signing
+
+The plugin supports setting up all your artifacts to be signed with GPG. Since that is a requirement
+for our default publishing target Maven Central, signing is enabled by default. For that please add the following 
+Gradle properties.
```
Comment (vanniktech) [inline README.md]:
```suggestion

```

File: README.md (hunk 3 above)
Comment (gabrielittner) [reply README.md]: Do you mean something other than `SONATYPE_NEXUS_USERNAME` and `SONATYPE_NEXUS_PASSWORD` which are mentioned in the next line?

Comment (vanniktech) [reply README.md]: Oh right they're both sharing the same name. Then never mind

**Pull request merged**